DancingGoat/Areas/Admin/Views/Shared/_MessagesPartial.cshtml

```diff
@@ -1,43 +1,44 @@
-﻿@{
-    const string imagePathPattern = "~/images/Admin/{0}.svg";
+﻿@using DancingGoat.Areas.Admin.Models
+@{
     var routeDataMessageBody = Context.Request.Query["MessageBody"].FirstOrDefault();
     var routeDataMessageType = Context.Request.Query["MessageType"].FirstOrDefault();
-    DancingGoat.Areas.Admin.Models.MessageModel viewBagMessage = null;
+    MessageModel viewBagMessage = null;
     string imageName = null;
-    string imagePath = null;
 
-    if (!string.IsNullOrEmpty(ViewBag.Message))
+    if (ViewBag.Message != null)
     {
         viewBagMessage = ViewBag.Message;
-        imageName = viewBagMessage.MessageType.ToString().ToLowerInvariant();
-        imagePath = string.Format(imagePathPattern, imageName);
+        imageName = CreateImageNameFromMessageType(viewBagMessage.MessageType);
     }
 
     if (!string.IsNullOrEmpty(routeDataMessageType))
     {
-        imageName = routeDataMessageType.ToString().ToLowerInvariant();
-        imagePath = string.Format(imagePathPattern, imageName);
+        imageName = routeDataMessageType.ToLowerInvariant();
     }
 }
 
-@{
-    void MessageImage(string source, string altText)
+@functions
+{
+    static string CreateImageNameFromMessageType(MessageType messageType)
+        => messageType.ToString().ToLowerInvariant();
+
+    void MessageImage(string imageName)
     {
-        <img src="@source" height="44" width="44" alt="@altText" />
+        <div class="user-message__img user-message__img--@imageName"></div>
     }
 }
```

Comment:
Usually, functions like these should be in helpers. However, they are somewhat tied to this very admin template, so it seems right to "let it slip".